rover-apps/common/src/main.cpp

```diff
@@ -0,0 +1,59 @@
+#include "AppConfig.h"
+#include "mbed.h"
+
+Thread periodic_10s_thread(osPriorityNormal1);
+Thread periodic_1s_thread(osPriorityNormal2);
+Thread periodic_100ms_thread(osPriorityNormal3);
+Thread periodic_10ms_thread(osPriorityNormal4);
+Thread periodic_1ms_thread(osPriorityNormal5);
+
+void periodic_10s(void) {
+  auto startTime = Kernel::Clock::now();
+  for (Module* module : gModules) {
+    module->periodic_10s();
+  }
+  ThisThread::sleep_until(startTime + 10s);
```

Comment:
we may want to add some logging for when we fail to hit deadlines. probs a future issue, but will help with debugging bringup for SAR